src/components/network/ImportNetwork.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import { RouteComponentProps } from 'react-router';
+import { UploadOutlined } from '@ant-design/icons';
+import styled from '@emotion/styled';
+import { Card, PageHeader, Spin, Upload } from 'antd';
+import { RcFile } from 'antd/lib/upload';
+import { usePrefixedTranslation } from 'hooks';
+import { useTheme } from 'hooks/useTheme';
+import { useStoreActions } from 'store';
+import { ThemeColors } from 'theme/colors';
+import { HOME } from 'components/routing';
+
+const Styled = {
+  Container: styled.div`
+    display: flex;
+    flex-direction: column;
+    height: 100%;
+  `,
+  PageHeader: styled(PageHeader)<{ colors: ThemeColors['pageHeader'] }>`
+    border: 1px solid ${props => props.colors.border};
+    border-radius: 2px;
+    background-color: ${props => props.colors.background};
+    margin-bottom: 10px;
+    flex: 0;
+  `,
+  Card: styled(Card)`
+    flex: 1;
+  `,
+  Dragger: styled(Upload.Dragger)`
+    flex: 1;
+  `,
+};
+
+const ImportNetwork: React.FC<RouteComponentProps> = () => {
+  const { navigateTo, notify } = useStoreActions(s => s.app);
+  const { importNetwork } = useStoreActions(s => s.network);
+  const { l } = usePrefixedTranslation('cmps.network.ImportNetwork');
+  const [importing, setImporting] = useState(false);
+
+  const doImportNetwork = (file: RcFile) => {
+    setImporting(true);
+
+    // we kick off the import promise, but don't wait for it
+    importNetwork(file.path)
+      .then(network => {
+        notify({ message: l('importSuccess', { name: network.name }) });
+        navigateTo(HOME);
+      })
+      .catch(error => {
+        notify({ message: l('importError', { file: file.name }), error });
+      })
+      .then(() => {
+        setImporting(false);
+      });
+
+    // return false to prevent the Upload.Dragger from sending the file somewhere
+    return false;
+  };
```

Comment:
This function is not tested in unit tests. One solution here could be to create a hidden element on the page that has a `onClick` set to this, and then assert on that in tests. Any thoughts on that?